packages/drivers/odsp-driver/src/getFileLink.ts

```diff
@@ -95,6 +105,46 @@ export async function getFileLink(
 	return fileLink;
 }
 
+/**
+ * Handles location redirection while fulfilling the getFilelink call.
+ * @param getToken - token fetcher to fetch the token.
+ * @param odspUrlParts - parts of odsp resolved url.
+ * @param logger - logger to send events.
+ * @returns Response from the API call.
+ * @alpha
+ */
+async function getFileLinkWithLocationRedirectionHandling(
+	getToken: TokenFetcher<OdspResourceTokenFetchOptions>,
+	odspUrlParts: IOdspUrlParts,
+	logger: ITelemetryLoggerExt,
+): Promise<string> {
+	// We can have chains of location redirection one after the other, so have a for loop
+	// so that we can keep handling the same type of error.
+	for (;;) {
```

Comment:
Should there be an escape clause in case of server error that repreatedly sends redirect response? For instance, set artificial max number of handled redirects for single item.

Reply:
No, there could be chain of redirections and we want to handle them all. There should not be more than a few. We do the same while loading and have not received any complaints.

Reply:
Added max retries.